### ISSUE TITLE
[Question] Upgrade Dependencies to fix Vulnerability

### ISSUE THREAD

**Issue by akrikos** [npm/gauge#135]
# What / Why
There's a Regular Expression Denial of Service vulnerability in the ansi-regex package in versions < 6.0.1: https://app.snyk.io/vuln/SNYK-JS-ANSIREGEX-1583908

Would it be possible to update gauge deps to use a patched version?

The dependencies from this package that use out of date versions that I see are:
string-width@2.0.0 > strip-ansi@3.0.1 > ansi-regex@2.1.1
strip-ansi@4.0.0 > ansi-regex@3.0.0

Fix version is https://github.com/chalk/ansi-regex/releases/tag/v6.0.1

## References
* https://app.snyk.io/vuln/SNYK-JS-ANSIREGEX-1583908

**Comment by akrikos** [comment]
I guess the question is whether or not gauge uses this on untrusted user input per the Fix release notes:
> You are only really affected if you run the regex on untrusted user input in a server context, which it's very unlikely anyone is doing, since this regex is mainly used in command-line tools.

**Comment by anjbur** [comment]
Chiming in as another voice that would appreciate this update! Is this something that will be fixed?

**Comment by massakam** [comment]
I'm also suffering from this issue and hope it will be fixed. Because `node-gyp` depends on `gauge`, the dependencies of native addon modules using `node-gyp` inevitably include a vulnerable version of `ansi-regex`.
```
node-gyp@7.1.0 -> npmlog@4.1.2 -> gauge@2.7.4 -> strip-ansi@3.0.1 -> ansi-regex@2.1.1
```

**Comment by stof** [comment]
@akrikos upgrading dependencies is still a good idea to maximize deduplication for `npm` dependencies (see #127) and to use maintained versions of the dependencies.

And looking at the code quicky, it seems like the strings passed to `string-width`, `strip-ansi` or `wide-align` are coming from the input received by `gauge`. And so `gauge` cannot know whether it is trusted or no (which means the safe behavior is to assume they are not)

**Comment by allanice001** [comment]
This vuln leaks all the up to bcrypt
https://github.com/kelektiv/node.bcrypt.js/issues/891

```
  ✗ Regular Expression Denial of Service (ReDoS) [High Severity][https://snyk.io/vuln/SNYK-JS-ANSIREGEX-1583908] in ansi-regex@5.0.0
    introduced by bcrypt@5.0.1 > @mapbox/node-pre-gyp@1.0.5 > npmlog@4.1.2 > gauge@2.7.4 > strip-ansi@3.0.1 > ansi-regex@2.1.1 and 157 other path(s)
  This issue was fixed in versions: 6.0.1
```

**Comment by t1m0thyj** [comment]
A patch update to `gauge@^2` would be appreciated. 🙏

Any package that includes `node-gyp` or `prebuild-install` as a dependency is affected by this vulnerability:
`node-gyp@8.2.0 -> npmlog@4.1.2 -> gauge@2.7.4 -> strip-ansi@3.0.1 -> ansi-regex@2.1.1`
`prebuild-install@6.1.4 -> npmlog@4.1.2 -> gauge@2.7.4 -> strip-ansi@3.0.1 -> ansi-regex@2.1.1`

**Comment by StudioMaX** [comment]
It seems that it is not so easy to fix this, since the problem is also in the direct dependency `wide-align` of gauge:
https://github.com/npm/gauge/blob/d345e195676bc9cc487337468255d737a4d57889/package.json#L33
```
gauge@~2.7.3 > wide-align@^1.1.0 > string-width@^1.0.1 > strip-ansi@^3.0.0 > ansi-regex@^2.0.0
```

But this [`wide-align` package](https://github.com/iarna/wide-align) has not been updated since May 2018.

**Comment by deepakvk** [comment]
Is there any update on this yet?

**Comment by t1m0thyj** [comment]
The wide-align package has been updated now 🎉 But other dependencies of gauge still need to be updated.

**Comment by StudioMaX** [comment]
I've created a [PR](https://github.com/npm/gauge/pull/136) with fixed dependencies in gauge 3.x. 
But only someone of the maintainers can update 2.x, since there is no branch for 2.x in this repository (it can be taken from commit https://github.com/npm/gauge/commit/1011abf6c2cb7ae89a3ee76fb447d3182d4e8d3a).

**Comment by opravil-jan** [comment]
Hi,

any progress in this issue, please?

Thanks
John

**Comment by mattolson** [comment]
This is bubbling up to grpc due to dependency on @mapbox/node-pre-gyp, which depends on gauge through npmlog. Any update on this?

**Comment by fredkilbourn** [comment]
I missed this issue and associated PR's here because I was originally looking at https://github.com/iarna/gauge which was linked from the npm website.

In any case I threw out my PR to address this in both locations:
https://github.com/iarna/gauge/pull/1
https://github.com/npm/gauge/pull/139

**Comment by wraithgar** [comment]
https://github.com/npm/gauge/pull/138

**Comment by fredkilbourn** [comment]
I suppose if maintainers are unceremoniously closing this issue we should move downstream to the packages using gauge to stop using it or put pressure upstream.

**Comment by stof** [comment]
@fredkilbourn this has been closed because #138 has upgraded the dependencies in 4.0.0

**Comment by fredkilbourn** [comment]
appreciate it thanks.  sorry for the snark in my last comment I'll delete it.
